src/test/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporterTest.java

```diff
@@ -442,43 +521,97 @@ private static BidRequest givenBidRequest(
                 .cur("USD")).build();
     }
 
-    private static SeatBid givenSeatBid(UnaryOperator<SeatBid.SeatBidBuilder> seatBidCostumizer) {
+    private static SeatBid givenSeatBid(UnaryOperator<SeatBid.SeatBidBuilder> seatBidCostumizer,
+                                        UnaryOperator<Bid.BidBuilder>... bidCustomizers) {
         return seatBidCostumizer.apply(SeatBid.builder()
-                .bid(Collections.singletonList(givenBid(bid -> bid)))
-                .seat("seat1")).build();
+                        .bid(givenBids(bidCustomizers))).build();
+    }
+
+    private static List<Bid> givenBids(UnaryOperator<Bid.BidBuilder>... bidCustomizers) {
+        return Arrays.stream(bidCustomizers)
+                .map(GreenbidsAnalyticsReporterTest::givenBid)
+                .toList();
     }
 
     private static Bid givenBid(UnaryOperator<Bid.BidBuilder> bidCustomizer) {
         return bidCustomizer.apply(Bid.builder()
-                .id("bid1")
                 .impid("adunitcodevalue")
-                .price(BigDecimal.valueOf(1.5))
                 .adm("<div>Ad Markup</div>")).build();
     }
 
     private static BidRejectionTracker givenBidRejectionTracker() {
         final BidRejectionTracker bidRejectionTracker = new BidRejectionTracker(
-                "seat2",
+                "seat3",
                 Set.of("adunitcodevalue"),
                 1.0);
         bidRejectionTracker.reject("imp1", BidRejectionReason.NO_BID);
         return bidRejectionTracker;
     }
 
+    private static ObjectNode givenPrebidBidderParamsNode() {
+        return givenPrebidObjectNode(
+                createBidderParams("seat1", params -> params
+                        .put("accountId", 1001)
+                        .put("siteId", 267318)
+                        .put("zoneId", 1861698)),
+                createBidderParams("seat2", params -> params
+                        .put("publisherId", 111)
+                        .put("adSlotId", 123456)),
+                createBidderParams("seat3", params -> params
+                        .put("placementId", 222))
+        );
+    }
+
+    private static ObjectNode givenPrebidObjectNode(UnaryOperator<ObjectNode>... bidderCustomizers) {
+        final ObjectNode bidderNode = mapper.createObjectNode();
+        Arrays.stream(bidderCustomizers).forEach(customizer -> customizer.apply(bidderNode));
+
+        final ObjectNode prebidNode = mapper.createObjectNode();
+        prebidNode.set("bidder", bidderNode);
+        return prebidNode;
+    }
+
+    private static UnaryOperator<ObjectNode> createBidderParams(
+            String bidder, UnaryOperator<ObjectNode> paramsCustomizer) {
+        return bidderNode -> {
+            final ObjectNode paramsNode = mapper.createObjectNode();
+            paramsCustomizer.apply(paramsNode);
+            bidderNode.set(bidder, paramsNode);
+            return bidderNode;
+        };
+    }
```

Comment:
I don't like the approach of returning operator instead of an object

especially when there is a single usage of `givenPrebidObjectNode` and you never really change the state of the object you build